library/src/blas1/rocblas_scal.hpp

```diff
@@ -0,0 +1,83 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#ifndef __SCAL_HPP__
+#define __SCAL_HPP__
+
+#include "handle.h"
+#include "logging.h"
+#include "rocblas.h"
+#include "scal_host.hpp"
+#include "utility.h"
+
+template <typename T, typename = T>
+constexpr char rocblas_scal_name[] = "unknown";
+template <>
+constexpr char rocblas_scal_name<float>[] = "rocblas_sscal";
+template <>
+constexpr char rocblas_scal_name<double>[] = "rocblas_dscal";
+template <>
+constexpr char rocblas_scal_name<rocblas_float_complex>[] = "rocblas_cscal";
+template <>
+constexpr char rocblas_scal_name<rocblas_double_complex>[] = "rocblas_zscal";
+template <>
+constexpr char rocblas_scal_name<rocblas_float_complex, float>[] = "rocblas_csscal";
+template <>
+constexpr char rocblas_scal_name<rocblas_double_complex, double>[] = "rocblas_zdscal";
```

Comment:
These need to be declared `static` if in a header file. `constexpr` template variables have external linkage by default in a retroactively-applied decision made earlier this year. We don't want to use unnamed namespaces in headers, so that leaves `static` (in C++17 the `constexpr` makes them implicitly `inline`, meaning they can be repeatedly defined if identically, but that doesn't apply to C++14).